src/Symfony/Bridge/Doctrine/Form/Type/DoctrineType.php

```diff
@@ -280,4 +280,9 @@ public function getParent()
     {
         return 'Symfony\Component\Form\Extension\Core\Type\ChoiceType';
     }
+
+    public function reset()
+    {
+        $this->choiceLoaders = array();
```

Comment:
You're throwing away all choice loaders. Wouldn't it be enough to iterate over all choice loaders and reset the `$choiceList` property of each of them?

Reply:
Looking at the code, I think it's fine this way: clearing DoctrineChoiceLoader::$choiceList would require more CPU here, but wouldn't save much on new iterations.